internal/services/contracts_events_consumer.go

```diff
@@ -689,6 +692,35 @@ func (c *ContractsEventsConsumer) dcnNewExpiration(e *ContractEventData) error {
 	return nil
 }
 
+// aftermarketDeviceAddressReset handles the event of the same name from the registry contract.
+func (c *ContractsEventsConsumer) aftermarketDeviceAddressReset(e *ContractEventData) error {
+	if e.ChainID != c.settings.DIMORegistryChainID || e.Contract != common.HexToAddress(c.settings.DIMORegistryAddr) {
+		return fmt.Errorf("aftermarket device address reset from unexpected source %d/%s", e.ChainID, e.Contract)
+	}
+
+	var args contracts.RegistryAftermarketDeviceAddressReset
+	if err := json.Unmarshal(e.Arguments, &args); err != nil {
+		return err
+	}
+
+	amd, err := models.AftermarketDevices(
+		models.AftermarketDeviceWhere.TokenID.EQ(utils.BigToDecimal(args.TokenId)),
+	).One(context.Background(), c.db.DBS().Reader)
+	if err != nil {
+		return err
+	}
+
+	amd.EthereumAddress = args.AftermarketDeviceAddress.Bytes()
+
+	return amd.Upsert(
+		context.Background(),
+		c.db.DBS().Writer,
+		true,
+		[]string{models.AftermarketDeviceColumns.EthereumAddress},
+		boil.Columns{Cols: []string{models.AftermarketDeviceColumns.EthereumAddress}},
```

Comment:
I think you want `boil.Whitelist`. And I think you may have to whitelist `updated_at` as well.